ddprof-lib/src/main/cpp/wallClock.cpp

```diff
@@ -161,49 +161,83 @@ void WallClockJVMTI::timerLoop() {
   // Attach to JVM as the first step
     VM::attachThread("Datadog Profiler Wallclock Sampler");
     auto collectThreads = [&](std::vector<ThreadEntry>& threads) {
-        jvmtiEnv* jvmti = VM::jvmti();
-        if (jvmti == nullptr) {
-            return;
-        }
-        JNIEnv* jni = VM::jni();
-
-        jint threads_count = 0;
-        jthread* threads_ptr = nullptr;
-        jvmti->GetAllThreads(&threads_count, &threads_ptr);
+      jvmtiEnv* jvmti = VM::jvmti();
+      if (jvmti == nullptr) {
+          return;
+      }
+      JNIEnv* jni = VM::jni();
 
-        bool do_filter = Profiler::instance()->threadFilter()->enabled();
-        int self = OS::threadId();
+      jint threads_count = 0;
+      jthread* threads_ptr = nullptr;
+      jvmti->GetAllThreads(&threads_count, &threads_ptr);
```

Comment:
can you comment a little on this part.
So we are getting jthread references, do we eventually always want to remove the local ref ? Or are there cases where we want to keep the local ref ?
Could some kind of RAII design, ensure that when we leave the function, we just delete all local refs to the jthreads ? (avoiding some of the branches bellow?)

Reply:
Added an extra comment